pkg/pb/synthetic_monitoring/checks.proto

```diff
@@ -133,6 +134,7 @@ message Probe {
   bool   deprecated        = 14 [(gogoproto.jsontag) = "deprecated"];                           // whether the probe has been deprecated
   double created           = 100 [(gogoproto.jsontag) = "created"];                             // time the probe was created
   double modified          = 101 [(gogoproto.jsontag) = "modified"];                            // last time modified
+  bool k6Enabled           = 102 [(gogoproto.jsontag) = "k6Enabled"];                           // probe has scripted checks enabled
```

Comment:
`ScriptedCheckEnabled` is sent by the Probe to the API to let the API know that this probe can run scripted checks.

Otherwise, the API should throw an error if the user assigns scripted checks to this probe.

Adding `k6Enabled` to the `Probe` object allows:
1. The FE to see which probes are eligible for scripted checks - in the future, we can maybe filter out ineligible probes when creating scripted checks.
2. Fetching this field from the DB - we call `toSM` on `db.probe`. This is used in [this PR](https://github.com/grafana/synthetic-monitoring-api/pull/815) so we can throw the API error properly.